include/sparrow_ipc/stream_file_serializer.hpp

```diff
@@ -16,15 +16,29 @@
 
 namespace sparrow_ipc
 {
+    /**
+     * @brief Represents a block entry in the Arrow IPC file footer.
+     *
+     * Each block describes the location and size of a record batch in the file.
+     */
+    struct record_batch_block
```

Comment:
`record_batch_block` name isn't really explicit regarding what it should be, maybe rename to something else to clarify? `record_batch_metadata` or `record_batch_info` or `record_batch_offsets` (preference for the latter)...